cmd/gogio/windowsbuild.go

```diff
@@ -264,53 +240,158 @@ func (f *windowsManifest) encode(w io.Writer) error {
+				// https://docs.microsoft.com/pt-br/windows/win32/menurc/string-str
+				newValue(valueText, "ProductVersion", v.VersionHuman),
+				newValue(valueText, "FileVersion", v.VersionHuman),
+				newValue(valueText, "FileDescription", v.Name),
+				newValue(valueText, "ProductName", v.Name),
+				// @TODO include more data: gogio must have some way to provide such information (like Company Name, Copyright...)
+			}),
+		}),
+		// https://docs.microsoft.com/pt-br/windows/win32/menurc/varfileinfo
+		newValue(valueBinary, "VarFileInfo", []io.WriterTo{
+			// https://docs.microsoft.com/pt-br/windows/win32/menurc/var-str
+			newValue(valueBinary, "Translation", uint32(page)<<16|uint32(v.Language)),
+		}),
+	})
+
+	// For some reason the ValueLength of the VS_VERSIONINFO must be the byte-length of the `windowsInfoValueFixed`:
+	t.ValueLength = 52
+
+	var verrsrc bufferCoff
+	if _, err := t.WriteTo(&verrsrc); err != nil {
 		return err
 	}
 
-	return template.Execute(w, f)
+	b.Coff.AddResource(16, 1, &verrsrc)
+
+	return nil
+}
+
+type windowsInfoValueFixed struct {
+	Signature      uint32
+	StructVersion  uint32
+	FileVersion    [2]uint32
+	ProductVersion [2]uint32
+	FileFlagMask   uint32
+	FileFlags      uint32
+	FileOS         uint32
+	FileType       uint32
+	FileSubType    uint32
+	FileDate       [2]uint32
+}
+
+func (v windowsInfoValueFixed) WriteTo(w io.Writer) (_ int64, err error) {
+	return 0, binary.Write(w, binary.LittleEndian, v)
+}
+
+type windowsInfoValue struct {
+	Length      uint16
+	ValueLength uint16
+	Type        uint16
+	Key         []byte
+	Value       []byte
+}
+
+func (v windowsInfoValue) WriteTo(w io.Writer) (_ int64, err error) {
+	// binary.Write doesn't support []byte inside struct.
+	if err = binary.Write(w, binary.LittleEndian, [3]uint16{v.Length, v.ValueLength, v.Type}); err != nil {
+		return 0, err
+	}
+	if _, err = w.Write(v.Key); err != nil {
+		return 0, err
+	}
+	if _, err = w.Write(v.Value); err != nil {
+		return 0, err
+	}
+	return 0, nil
+}
+
+const (
+	valueBinary uint16 = 0
+	valueText   uint16 = 1
+)
+
+func newValue(valueType uint16, key string, input interface{}) windowsInfoValue {
+	v := windowsInfoValue{
+		Type: valueType,
+		Key:  stringBytes(key, 6),
+	}
+
+	v.Length = 6 + uint16(len(v.Key))
+
+	switch in := input.(type) {
+	case string:
+		v.Value = stringBytes(in, 0)
+		v.ValueLength = uint16(len(v.Value) / 2)
+	case []io.WriterTo:
+		var buff bytes.Buffer
+		for k := range in {
+			if _, err := in[k].WriteTo(&buff); err != nil {
+				panic(err)
+			}
+		}
+		v.Value = buff.Bytes()
+	default:
+		var buff bytes.Buffer
+		if err := binary.Write(&buff, binary.LittleEndian, in); err != nil {
+			panic(err)
+		}
+		v.ValueLength = uint16(buff.Len())
+		v.Value = buff.Bytes()
+	}
+
+	v.Length += uint16(len(v.Value))
+
+	return v
+}
+
+var utf16encoder = unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM).NewEncoder()
+
+func stringBytes(s string, previous int) []byte {
```

Comment:
Rename stringBytes to utf16Encode, and comment that the encoded string is null-terminated.